File: tests/calamares_status.py
#!/usr/bin/env python3
"""
Test script to verify Calamares monitoring functionality
"""

import sys
import time
from pathlib import Path

# Add the project root to the path
sys.path.insert(0, str(Path(__file__).parent))

from utils.extras import is_calamares_running, _calamares_running, _calamares_lock
import threading

def test_calamares_status():
    """Test the Calamares status checking functionality"""
    print("Testing Calamares monitoring functionality...")
    
    # Test initial state
    print(f"Initial Calamares status: {is_calamares_running()}")
    assert not is_calamares_running(), "Calamares should not be running initially"
    
    # Simulate Calamares starting
    print("Simulating Calamares start...")
    with _calamares_lock:
        global _calamares_running
        _calamares_running = True
    
    print(f"Calamares status after start: {is_calamares_running()}")
    assert is_calamares_running(), "Calamares should be running after start"
    
    # Simulate Calamares stopping
    print("Simulating Calamares stop...")
    with _calamares_lock:
        _calamares_running = False
    
    print(f"Calamares status after stop: {is_calamares_running()}")
    assert not is_calamares_running(), "Calamares should not be running after stop"
    
    print("All tests passed! Calamares monitoring functionality works correctly.")

if __name__ == "__main__":
    test_calamares_status()

File: tests/calamares_status.cpp
#include "utils/Extras.h"
#include <iostream>
#include <stdexcept>

int main() {
    using namespace std;
    cout << "Testing Calamares monitoring functionality..." << endl;

    // Initial state
    bool initial = Extras::isCalamaresRunning();
    cout << "Initial Calamares status: " << (initial ? "true" : "false") << endl;
    if (initial) {
        cerr << "Expected Calamares not running initially" << endl;
        return 1;
    }

    // Simulate start
    cout << "Simulating Calamares start..." << endl;
    Extras::setCalamaresRunningForTest(true);
    bool afterStart = Extras::isCalamaresRunning();
    cout << "Calamares status after start: " << (afterStart ? "true" : "false") << endl;
    if (!afterStart) {
        cerr << "Expected Calamares running after start" << endl;
        return 2;
    }

    // Simulate stop
    cout << "Simulating Calamares stop..." << endl;
    Extras::setCalamaresRunningForTest(false);
    bool afterStop = Extras::isCalamaresRunning();
    cout << "Calamares status after stop: " << (afterStop ? "true" : "false") << endl;
    if (afterStop) {
        cerr << "Expected Calamares not running after stop" << endl;
        return 3;
    }

    cout << "All tests passed! Calamares monitoring functionality works correctly." << endl;
    return 0;
}
